### ISSUE TITLE
The Process of Obtaining the StepFunctions ARN fails in V.4

### ISSUE THREAD

**Issue by Hiroki-Aoki** [serverless/serverless#12838]
### Issue description

We are deploying using the following serverless.yml. No error occurs in V.3, but an error is encountered in V.4.

- serverless.yml

```yml
functions:
  - ${file(./functions_common.yml)}

stepFunctions:
  stateMachines: ${file(./state_machines.yml)}
```

- functions_common.yml

```yml
StepfunctionsAAAA:
  environment:
    stateMachineBBBBARN: !Ref ${self:stepFunctions.stateMachines.BBBB.name}
```

- state_machines.yml

```yml
BBBB:
  name: StateMachineBBBB
```
 
Error encountered:

```
Error: Cannot resolve '${self:stepFunctions.stateMachines.BBBB.name}' variable at 'functions.0.StepfunctionsAAAA.environment.stateMachineBBBBARN.Ref'. None of the variables returned a value and no default value was provided. Please check your variable definitions or provide a default value.
```

However, if we swap the order of the functions section and stepFunctions section in the serverless.yml, the error does not occur. Is this due to a change in specifications from V.3 to V.4?

**Comment by czubocha** [comment]
Hey @Hiroki-Aoki, thanks for reporting this. The fix will be included in the next release.

**Comment by czubocha** [comment]
Hey @Hiroki-Aoki, this has been addressed in version 4.4.4, which is now live. Feel free to try it out and let us know if you run into any issues.

**Comment by Hiroki-Aoki** [comment]
@czubocha Thanks for fix it.
